cdap-watchdog/src/main/java/co/cask/cdap/logging/read/KafkaLogReader.java

```diff
@@ -49,7 +49,6 @@
 
   private final BrokerService brokerService;
   private final String topic;
-  private final LoggingEventSerializer serializer;
```

Comment:
This is used by a KafkaCallback, which will be used to serialize the kafka message for each fetch request, and each `getLogNext` or `getLogPrev`will do a kafka fetch. I think each fetch request should use a new serializer since this is not thread safe.

Reply:
What @anwar6953 meant was to use a `ThreadLocal<LoggingEventSerializer>` field instead, essentially it creates one instance per thread (hence no concurrent access to the same instance), while the same instance can be reused by the same HTTP handler thread.